### DISCUSSION TITLE
Maximum bigint vs JS Number.MAX_SAFE_INTEGER

### DISCUSSION THREAD

**Discussion by Llewdur** [livewire/livewire#5124] [1. Help] [answered]
Hi there,

I have been battling this issue for weeks now. Will you please help.

Whenever the Livewire component contains data within the bigint range but exceeding the JS MAX_SAFE_INTEGER range, Livewire is returning "Livewire encountered corrupt data when trying to hydrate the ... component"

A simple example component:
```
class Bigint extends Component
{
    public array $state = [
        "article_id" => 9123456890123456789, // Works for 9123456890123456 but fails on 9123456890123456789
    ];

    public function mount()
    {
    }

    public function submit()
    {
        dd(__CLASS__, __LINE__,);
    }

    public function render()
    {
        return view('livewire.bigint');
    }
}
```

with a simple view:
```
<div>

    <form wire:submit.prevent="submit">
        <input type="text" name="foo">

        <button>Submit</button>
    </form>

</div>
```

Setting article_id to a smaller integer works fine but anything approaching the maximum bigint size fails
9123456890123456 => works
9123456890123456789 => fails

![image](https://user-images.githubusercontent.com/13761739/180629334-34ee5e1e-1fd9-4c82-8b9b-792dfcb4e1e3.png)

Please will you help me get around this issue

**Comment by joshhanley** [comment, marked as answer]
@Llewdur as you already suggested, you are running into Javascript's max integer size.

The only way to work around that is to convert any large integers into strings. Otherwise, when your int is sent to the front end, javascript will change the number due to the limit, which triggers Livewire's security checksum as it was not the same number that was sent.

See my comment here for more details https://github.com/livewire/livewire/discussions/4743#discussioncomment-2277096

Hope this helps!

**Reply by Llewdur** [reply]
hi @joshhanley - thank you for your help. Casting it to strings solved the issue but the code looks messy as we have to loop through all the ElasticSearch articles to cast the IDs before rendering the search results.

Im hoping Livewire can do the casting at some stage as Laravel migrations are using unsigned bigint IDs by default

Thanks again for the solution though

**Reply by joshhanley** [reply]
@Llewdur no worries! 🙂

Yeah it's not ideal but it works.

Someone has tried before, but the implementation wasn't great. Have a look at Caleb's response and suggestion here https://github.com/livewire/livewire/discussions/3061#discussioncomment-892039
